techmatters/flex-plugins
# CHI-970: Dedup form def data

Primary reviewer: @acedeywin 

## Description

* Added an option 'ReferenceData.json' form definition file where data referenced from select and dependent select dropdowns can be referenced
* Updated loadDefinitions to expand their select dropdowns from references to inline options
* Improved tests around form definition validation to check basic shape of options lists
* Refactored ph-v1 form definitions so that the province, municipality and district dropdowns on the Child Information, Household and Perpetrator forms reference the same shared set of options in ReferenceData.json

### Checklist
- N/A Strings are localized

### Related Issues

CHI-970

### Verification steps

* Either locally run or deploy a version of flex 2 pointing at the PH_STG backend
* Fill out an offline contact, checking several options for provinces, municipalities & districts
* Add a case with a household & perpatrator, again checking the same dropdowns
* Verify dependent selects on Aselo Development to ensure they still work
